app/Helpers/GitHandler.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Helpers;
+
+class GitHandler
+{
+    private $message;
+    
+    public function fetchAndPull()
+    {
+        $branch = env('GIT_BRANCH', 'master');
+
+        exec("git fetch origin $branch", $output, $return_var);
+
+        if ($return_var === 0) {
+            $this->message = "Fetched updates from origin/$branch.";
+            exec("git reset --hard FETCH_HEAD", $output, $return_var);
+            if ($return_var === 0) {
+                $this->message = "Hard reset to latest updates from origin/$branch.";
+                exec("git clean -df", $output, $return_var);
+                if ($return_var === 0) {
+                    $this->message = "Cleaned up untracked files and directories.";
+                } else {
+                    $this->message = "Error cleaning untracked files and directories.";
+                }
+            } else {
+                $this->message = "Error resetting to latest updates from origin/$branch.";
+            }
+        } else {
+            $this->message = "Error fetching updates from origin/$branch.";
+        }
+
+        return $this->message;
```

Comment:
this needs some refactoring

Reply:
Thank you for contributions🙏🎉

Reply:
@Zakarialabib np!

I'd start of by creating an abstraction layer around the command you want to call